### PR TITLE
Fix numbering list triggers

### PR DESCRIPTION
**Pull request by juliaroldi**
Ensure list will be trigger only by the right combination of characters. 
listBugs

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts [hunk 1]
```diff
@@ -109,9 +112,17 @@ const identifyNumberingListType = (numbering: string): NumberingListType | null
  * @param textBeforeCursor The trigger character
  * @returns The style of a numbering list triggered by a string
  */
-export default function getAutoNumberingListStyle(textBeforeCursor: string): NumberingListType {
+export default function getAutoNumberingListStyle(
+    textBeforeCursor: string,
+    num?: number
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts:117]: add comment for this parameter

File: packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts [hunk 2]
```diff
@@ -90,16 +90,19 @@ const DecimalsTypes: Record<number, number> = {
...
+    const separatorCharacter = isDoubleParenthesis
+        ? Character.DoubleParenthesis
+        : characters[numbering[1]];
     // if separator is not valid, no need to check if the number is valid.
-    if (char) {
+    if (separatorCharacter) {
         const number = numbering.length === 3 ? numbering[1] : numbering[0];
-        const numberingType = identifyNumberingType(number);
-        return numberingType ? numberingListTypes[numberingType](char) : null;
+        const numberingType = num ? identifyNumberingType(number) : numberingTriggers[number];
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts:104]: How to do you want to handle num == 0?

Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts]: 0 can treated as undefined, if num === 0 it means the list must start from beginning, so we should only check for a, A, i, I and 1, them numberingTrigger is enough

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getAutoNumberingListStyle.ts]: ok

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**